desktop/cmp/treemap/TreeMapModel.js

```diff
@@ -0,0 +1,405 @@
+     *      If not provided, by default will expand / collapse a record when using a GridModel.
+     * @param {(boolean|TreeMapModel~tooltipFn)} [c.tooltip] - 'true' to use the default tooltip renderer, or a custom
+     *      tooltipFn which returns a string output of the node's value.
+     * @param {(Element|string)} [c.emptyText] - Element/text to render if TreeMap has no records.
+     *      Defaults to null, in which case no empty text will be shown.
+     */
+    constructor({
+        store,
+        gridModel,
+        maxNodes = 1000,
+        highchartsConfig,
+        labelField = 'name',
+        valueField = 'value',
+        heatField = 'value',
+        maxDepth,
+        algorithm = 'squarified',
+        colorMode = 'linear',
+        onClick,
+        onDoubleClick,
+        tooltip = true,
+        emptyText,
+        filter
+    } = {}) {
+        this.gridModel = gridModel;
+        this.store = store ? store : gridModel ? gridModel.store : null;
+        throwIf(!this.store,  'TreeMapModel requires either a Store or a GridModel');
+        this.maxNodes = maxNodes;
+
+        this.highchartsConfig = highchartsConfig;
+        this.labelField = labelField;
+        this.valueField = valueField;
+        this.heatField = heatField;
+        this.maxDepth = maxDepth;
+
+        throwIf(!['sliceAndDice', 'stripes', 'squarified', 'strip'].includes(algorithm), `Algorithm "${algorithm}" not recognised.`);
+        this.algorithm = algorithm;
+
+        throwIf(!['linear', 'balanced', 'none'].includes(colorMode), `Color mode "${colorMode}" not recognised.`);
+        this.colorMode = colorMode;
+
+        this.onClick = withDefault(onClick, this.defaultOnClick);
+        this.onDoubleClick = withDefault(onDoubleClick, this.defaultOnDoubleClick);
+        this.tooltip = tooltip;
+        this.emptyText = emptyText;
+
+        this._filter = filter;
+
+        this.addReaction({
+            track: () => [
+                this.store.rootRecords,
+                this.expandState,
+                this.colorMode,
+                this.labelField,
+                this.valueField,
+                this.heatField,
+                this.maxDepth
+            ],
+            run: ([rawData]) => this.data = this.processData(rawData)
+        });
+    }
+
+    @computed
+    get total() {
+        return sumBy(this.data, it => {
+            // Only include root records that pass the filter
+            if (it.parent || (this._filter && !this._filter(it.record))) return 0;
+            return it.value;
+        });
+    }
+
+    @computed
+    get valueFieldLabel() {
+        const field = this.store.fields.find(it => it.name === this.valueField);
+        return field ? field.label : this.valueField;
+    }
+
+    @computed
+    get heatFieldLabel() {
+        const field = this.store.fields.find(it => it.name === this.heatField);
+        return field ? field.label : this.heatField;
+    }
+
+    @computed
```

Comment:
do we get anything out of the computed for selectedIds and expandState?  Or is this just for consistentcy?

Reply:
I know we've discussed this before, how if the getter is a trivial transform of an observable the @computed doesn't add that much in terms of efficiency. And that the reaction / component render will detect any observables accessed during its run (however transitively?) and respond to their changes automatically.

That said, I still favor this setup to make it explicit that these are observable getters, without needing to look at their contents too closely to work out if their internals are observables. 

The consistency also seems helpful, and avoids judgement calls around "this implementation used to be trivial. but now has gotten a bit more complicated, should it get the decorator now?"